OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2117,6 +2129,10 @@ private static OSNotificationOpenResult generateOsNotificationOpenResult(JSONArr
    }
 
    private static void fireNotificationOpenedHandler(final OSNotificationOpenResult openedResult) {
+      // TODO: Is there a reason we need the opened handler to be fired from main thread?
```

Comment:
We do this so app devs can run their UI update code without having to worry about getting themselves on the main thread.